### PR TITLE
C2s testing -- first step

### PR DESCRIPTION
**Pull request by pzel**
This is an initial pull request aimed at easing `ejabberd_c2s` refactoring.
Three things have been achieved:
1) Create separate test suite (`apps/ejabberd2/tests/ejabberd_c2s_SUITE.erl`).
2) Move `c2s`-related defines and record declarations to `src/ejabberd.hrl`. This is an internal include and still needs work, but it's a first step towards better organization.
3) Extract all `c2s` dependencies and launch them in `ejabberd_c2s_SUITE_mocks:[setup/0, teardown/0]`

### CODE REVIEW

File: apps/ejabberd/src/ejabberd_c2s.hrl [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                stream_mgmt_out_acked = 0,
+                stream_mgmt_buffer = [],
+                stream_mgmt_buffer_size = 0,
+                stream_mgmt_buffer_max = ?STREAM_MGMT_CACHE_MAX,
+                stream_mgmt_ack_freq = ?STREAM_MGMT_ACK_FREQ,
+                stream_mgmt_resume_timeout = ?STREAM_MGMT_RESUME_TIMEOUT,
+                stream_mgmt_resume_tref,
+                stream_mgmt_constraint_check_tref
+                }).
+-type aux_key() :: atom().
```
Comment (lavrin) [inline apps/ejabberd/src/ejabberd_c2s.hrl]: I don't think putting types in .hrl files is a good idea due to possible name clashes. Namespacing options in Erlang are limited, but let's use them when possible ;)

**Comment by pzel** [comment]
Your sensible reservation was pre-empted in the commit message:

 >  2) Move c2s-related defines and record declarations to src/ejabberd.hrl. This is an internal include and still needs work, but it's a first step towards better organization.

The types & various other concerns will definitely be separated, but I don't want to fall in the trap of kitchen-sink pull requests.

**Comment by lavrin** [comment]
Just sayin' :)

File: apps/ejabberd/src/ejabberd_c2s.erl [hunk 2]
```diff
@@ -3100,208 +2973,3 @@ defer_resource_constraint_check(#state{stream_mgmt_constraint_check_tref = undef
     State#state{stream_mgmt_constraint_check_tref = TRef};
 defer_resource_constraint_check(State)->
     State.
-
--ifdef(TEST).
```
Comment (michalwski) [inline apps/ejabberd/src/ejabberd_c2s.erl]: Remember to resurrect this tests in next steps.

Comment (pzel) [inline apps/ejabberd/src/ejabberd_c2s.erl]: Certainly!

**Pull request merged**